client_src/js/lib/functions/ui.js

```diff
@@ -89,4 +89,17 @@ export default class UI {
       });
     });
   }
+  static removeLoaderFromElementInnerHTML(element) {
+    if (element.innerHTML.includes("spinner")) {
+      element.innerHTML = element.innerHTML.replace(
+        '<i class="fas fa-spinner fa-spin"></i>',
+        ""
+      );
+    }
+  }
+  static addLoaderToElementsInnerHTML(element) {
+    if (!element.innerHTML.includes("spinner")) {
+      element.innerHTML += ' <i class="fas fa-spinner fa-spin"></i>';
+    }
+  }
```

Comment:
Just tow new UI functions that I use quite a lot. Rest of client can start utilising these more but will migrate in another PR.

Reply:
The remove from loader was also causing the listener to disappear which is something I'll have to keep in mind. Initially there was no check if "spinner" was included in the innerHTML so it was replacing nothing basically. That was causing the listener to disappear which I had no idea was an effect of that.